.github/scripts/generate-index-markdown.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# Output file name
+output_file="docs/index.md"
+
+# Function to extract the title from a Markdown file
+extract_title() {
+    title=$(grep -E "^# (.*)" "$1" | sed -E "s/^# (.*)/\1/")
+    echo "$title"
+}
+
+# Function to generate the list of benchmark tests
+generate_tests_list() {
+    auth_tests_list=""
+    secret_tests_list=""
+    system_tests_list=""
+    for test_file in $(ls -1 docs/tests/* | sort); do
+        if [[ -f "$test_file" ]]; then
+            test_name=$(basename "$test_file")
+            test_link="[$(extract_title "$test_file")]($test_file)"
+            if [[ "$test_name" == *"auth"* ]]; then
+                auth_tests_list+="\n- $test_link"
+            elif [[ "$test_name" == *"secret"* ]]; then
+                secret_tests_list+="\n- $test_link"
+            elif [[ "$test_name" == *"system"* ]]; then
+                system_tests_list+="\n- $test_link"
+            fi
+        fi
+    done
+    echo "\n### Auth Benchmark Tests$auth_tests_list\n\n### Secret Benchmark Tests$secret_tests_list\n\n### System Tests$system_tests_list"
+}
+
+# Create the Markdown content
+markdown_content="# Vault Benchmark
```

Comment:
Can we instead take in a markdown file as a template instead of updating the script?

Reply:
Sure thing!